bot.js

```diff
@@ -682,6 +683,116 @@ function command(message) {
+            if (detailsObject.internalKey == null) {
+                // If we haven't been told how to, log an error and use the passed key instead.
+                log.error(
+                    "Detailed helptext spec for command " +
+                        target +
+                        " does not include an internalKey!"
+                );
+                response = "> **The " + target + " command**\n";
+            } else {
+                response =
+                    "> **" +
+                    config.commands[detailsObject.internalKey] +
+                    "**\n";
+            }
+
+            // Now, the subtitle.
+            if (detailsObject.subtitle == null) {
+                // If there is no subtitle, log a warning.
+                log.warning(
+                    "Detailed helptext spec for command " +
+                        target +
+                        " does not include a subtitle!"
+                );
+                response += "> \n";
+            } else {
+                response += "> " + detailsObject.subtitle + "\n> \n";
+            }
+
+            // And finally, the body.
+            if (detailsObject.lines == null) {
+                // If there is no body, log an error.
+                log.error(
+                    "Detailed helptext spec for command " +
+                        target +
+                        " does not include a lines array!"
+                );
+                response += "> I don't really have anything to say here.";
+            } else {
+                // If there is a body, process its lines:
+                //  - Format in any command prompts it requests
+                //  - Put the whole thing in blockquotes
+                //  - Add newlines after each line
+                response += detailsObject.lines
+                    .map(line => {
+                        let result = "";
+                        // Handle any %cmd syntax (references to a command's prompting message)
+                        const terms = line.split("%");
+                        // If there's no chance of any such syntax, skip the loop
+                        if (terms.length == 1) {
+                            result = line;
+                        } else {
+                            // Take everything before the first %, and add it to result
+                            result += terms[0];
+
+                            // For every term that started with a %...
+                            terms.splice(0, 1);
+                            for (const term of terms) {
+                                // Grab up to the first word boundary
+                                const firstWord = term.split(/\b/)[0];
+
+                                // If the first word is a command's internal name...
+                                if (
+                                    Object.keys(config.commands).includes(
```

Comment:
This is one of the cases where I am *really* not a fan of Prettier's chosen formatting...